.document

```diff
@@ -2,5 +2,6 @@ History.txt
 LICENSE.txt
 README.txt
 RI.txt
+RI.md
```

Comment:
👋🏻  You identified this document's primary audience is people new to the `ri` command, and they would find this document mostly through GitHub. But, when rendering the document in GitHub, there are lots of weird and missing formatting artifacts coming through that make it challenging to read:

* Document's title header is not rendered correctly 
* Links not being rendered correctly, i.e., `{Ruby online documentation}[https://docs.ruby-lang.org/en/master]:`
* Lots of use of links using `rdoc-ref`, but these are not rendered on GitHub. Does this mean the document is intended to be read through `ri`?  How do i do that?

Perhaps it would be helpful to either commit to Markdown and it's ways of formatting, or stick with the RDoc way and continue to use `ri.rdoc`?

Reply:
> Perhaps it would be helpful to either commit to Markdown and it's ways of formatting, or stick with the RDoc way and continue to use ri.rdoc?

I'd support committing to markdown format and will be happy to help with that conversion as I did [something similar](https://github.com/ruby/irb/pull/890) for IRB just not too long ago 🙂

Reply:
> 👋🏻 You identified this document's primary audience is people new to the `ri` command, and they would find this document mostly through GitHub. But, when rendering the document in GitHub, there are lots of weird and missing formatting artifacts coming through that make it challenging to read:
> 
>     * Document's title header is not rendered correctly
> 
>     * Links not being rendered correctly, i.e., `{Ruby online documentation}[https://docs.ruby-lang.org/en/master]:`
> 
>     * Lots of use of links using `rdoc-ref`, but these are not rendered on GitHub. Does this mean the document is intended to be read through `ri`?  How do i do that?
> 
> 
> Perhaps it would be helpful to either commit to Markdown and it's ways of formatting, or stick with the RDoc way and continue to use `ri.rdoc`?

@colby-swandale, is there a way I can view this new file as it's rendered on GitHub?  I'd like to see the problems there firsthand.

Also, not sure what you mean by 'commit to markdown'; the new page is `.md` -- a markdown page.  Maybe some of the text has an `rdoc` accent of which I'm not aware?

Reply:
> @colby-swandale, is there a way I can view this new file as it's rendered on GitHub? I'd like to see the problems there firsthand.

Sure! The current version is at: https://github.com/ruby/rdoc/blob/21505b91dd770e31c485205ac74ed44ef832432a/RI.md

> Also, not sure what you mean by 'commit to markdown';

The page is Markdown, but the document is utilising RDoc ways of doing things, particularly around links. When Github renders the document, the links are being rendered as raw text:

<img width="846" alt="Screenshot 2024-05-02 at 7 50 40 AM" src="https://github.com/ruby/rdoc/assets/996377/e19966d2-c5a1-477e-bdb7-6eb351d5fd08">

---

Links in Markdown need to be denoted with `[<text>](<link>)` instead of `{<text>}[<link>]` as per the [Markdown Spec](https://www.markdownguide.org/basic-syntax/#links)